Sources/OfflineEntitlements/PurchasedProductsFetcher.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchasedProductsFetcher.swift
+//
+//  Created by Andrés Boedo on 3/17/23.
+
+import Foundation
+import StoreKit
+
+
+/// This struct should have all the information we need from StoreKit  to create EntitlementInfo from a StoreKit 2 transaction.
+/// Other fields from other places might be needed.
+/// If not, then consider just having a constructor that creates EntitlementInfo from a Storekit.Transaction instead and using
+/// a Factory pattern to store the logic in one place
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+struct PurchasedSK2Product {
+    let productIdentifier: String
+    let expirationDate: Date?
+    let periodType: PeriodType
+
+    init(from transaction: StoreKit.Transaction) {
+        self.productIdentifier = transaction.productID
+        self.expirationDate = transaction.expirationDate
+        if let offerType = transaction.offerType {
+            switch offerType {
+            case .code, .promotional:
+                self.periodType = .intro
+            case .introductory:
+                // note: this isn't necessarily accurate, but there's no field in SK2 to
+                // tell us whether this is a free trial after all, so it's a best guess.
+                // since free trials are much more common than intro pricing, we're going with
+                // trial
+                self.periodType = .trial
+            default:
+                self.periodType = .normal
+            }
+        } else {
+            self.periodType = .normal
+        }
+    }
+}
+
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+struct PurchasedProductsManager {
+
+    func fetchPurchasedProducts() async throws -> [PurchasedSK2Product] {
```

Comment:
This would probably be `static`, right?

Reply:
I went with instance out of sheer habit. Based on the env detector there, though, it looks like current style is protocol + static + initializer with default param? Happy to go with that instead

Reply:
Oh wait, this is a "manager" ignore my comment. I was thinking this was a product type.

Reply:
so... to be clear, do you like this style better than the other one? I'm happy to go with either.

Reply:
Yeah this makes sense 👍🏻 
Both are the same thing TBH, but this makes mocking a bit easier.

Reply:
Actually this is a `struct`, so it can't be mocked. So might as well make this `static`. But this is fine for now, let's see how we need to use this and/or mock it and then we can change it.